libs/lb/lb-rs/src/shared/account.rs

```diff
@@ -24,6 +29,94 @@ impl Account {
     pub fn public_key(&self) -> PublicKey {
         PublicKey::from_secret_key(&self.private_key)
     }
+
+    pub fn get_phrase(&self) -> SharedResult<[&'static str; 24]> {
+        let key = self.private_key.serialize();
+        let key_bits = key.iter().fold(String::new(), |mut out, byte| {
+            let _ = write!(out, "{:08b}", byte);
+            out
+        });
+
+        let checksum: String =
+            sha2::Sha256::digest(&key)
+                .into_iter()
+                .fold(String::new(), |mut out, byte| {
+                    let _ = write!(out, "{:08b}", byte);
+                    out
+                });
+
+        let checksum_last_4_bits = &checksum[..4];
+        let combined_bits = format!("{}{}", key_bits, checksum_last_4_bits);
+
+        let mut phrase: [&str; 24] = Default::default();
+
+        for (i, chunk) in combined_bits
+            .chars()
+            .collect::<Vec<_>>()
+            .chunks(11)
+            .enumerate()
+        {
+            let index =
+                u16::from_str_radix(&chunk.iter().collect::<String>(), 2).map_err(|_| {
+                    SharedErrorKind::Unexpected(
+                        "could not parse appropriate private key bits into u16",
+                    )
+                })?;
+            let word = bip39_dict::ENGLISH.lookup_word(bip39_dict::MnemonicIndex(index));
+
+            phrase[i] = word;
+        }
+
+        Ok(phrase)
+    }
+
+    pub fn phrase_to_private_key(phrases: [&str; 24]) -> SharedResult<SecretKey> {
+        let mut combined_bits = phrases
+            .iter()
+            .map(|word| {
+                bip39_dict::ENGLISH
+                    .lookup_mnemonic(word)
+                    .map(|index| format!("{:011b}", index.0))
+            })
+            .collect::<Result<String, _>>()
+            .map_err(|_| SharedErrorKind::KeyPhraseInvalid)?;
+
+        if combined_bits.len() != 264 {
+            return Err(SharedErrorKind::Unexpected("the number of bits after translating the phrase does not equal the expected amount (264)").into());
+        }
+
+        for _ in 0..4 {
+            combined_bits.remove(253);
```

Comment:
The 23 word ends at index 252. On 253, we have the addition 0s we don't need, 4 bits of them, so we must remove them.

Honestly though, maybe I don't need to remove them. I'll check that right now.